test/e2e/dns.go

```diff
@@ -278,114 +300,167 @@ func reverseArray(arr []string) []string {
 var _ = framework.KubeDescribe("DNS", func() {
 	f := framework.NewDefaultFramework("dns")
 
-	It("should provide DNS for the cluster [Conformance]", func() {
-		// All the names we need to be able to resolve.
-		// TODO: Spin up a separate test service and test that dns works for that service.
-		namesToResolve := []string{
-			"kubernetes.default",
-			"kubernetes.default.svc",
-			"kubernetes.default.svc.cluster.local",
-			"google.com",
-		}
-		// Added due to #8512. This is critical for GCE and GKE deployments.
-		if framework.ProviderIs("gce", "gke") {
-			namesToResolve = append(namesToResolve, "metadata")
-		}
-		hostFQDN := fmt.Sprintf("%s.%s.%s.svc.cluster.local", dnsTestPodHostName, dnsTestServiceName, f.Namespace.Name)
-		hostEntries := []string{hostFQDN, dnsTestPodHostName}
-		wheezyProbeCmd, wheezyFileNames := createProbeCommand(namesToResolve, hostEntries, "", "wheezy", f.Namespace.Name)
-		jessieProbeCmd, jessieFileNames := createProbeCommand(namesToResolve, hostEntries, "", "jessie", f.Namespace.Name)
-		By("Running these commands on wheezy:" + wheezyProbeCmd + "\n")
-		By("Running these commands on jessie:" + jessieProbeCmd + "\n")
-
-		// Run a pod which probes DNS and exposes the results by HTTP.
-		By("creating a pod to probe DNS")
-		pod := createDNSPod(f.Namespace.Name, wheezyProbeCmd, jessieProbeCmd, true)
-		validateDNSResults(f, pod, append(wheezyFileNames, jessieFileNames...))
-	})
+	var _ = Describe("Service", func() {
+		It("should provide DNS for the cluster [Conformance]", func() {
+			// All the names we need to be able to resolve.
+			// TODO: Spin up a separate test service and test that dns works for that service.
+			namesToResolve := []string{
+				"kubernetes.default",
+				"kubernetes.default.svc",
+				"kubernetes.default.svc.cluster.local",
+				"google.com",
+			}
+			// Added due to #8512. This is critical for GCE and GKE deployments.
+			if framework.ProviderIs("gce", "gke") {
+				namesToResolve = append(namesToResolve, "metadata")
+			}
+			hostFQDN := fmt.Sprintf("%s.%s.%s.svc.cluster.local", dnsTestPodHostName, dnsTestServiceName, f.Namespace.Name)
+			hostEntries := []string{hostFQDN, dnsTestPodHostName}
+			wheezyProbeCmd, wheezyFileNames := createProbeCommand(namesToResolve, hostEntries, "", "wheezy", f.Namespace.Name)
+			jessieProbeCmd, jessieFileNames := createProbeCommand(namesToResolve, hostEntries, "", "jessie", f.Namespace.Name)
+			By("Running these commands on wheezy:" + wheezyProbeCmd + "\n")
+			By("Running these commands on jessie:" + jessieProbeCmd + "\n")
+
+			// Run a pod which probes DNS and exposes the results by HTTP.
+			By("creating a pod to probe DNS")
+			pod := createDNSPod(f.Namespace.Name, wheezyProbeCmd, jessieProbeCmd, true)
+			validateDNSResults(f, pod, append(wheezyFileNames, jessieFileNames...))
+		})
+
+		It("should provide DNS for services [Conformance]", func() {
```

Comment:
I think that's fine. Even without this we already have something like: "DNS should provide DNS for services"? We are just more specific now where we say "DNS {Service, Ingress} should provide DNS for services"?